### ISSUE TITLE
Review impact of collection expressions on existing rules

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#4007]
https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions

**Comment by DoctorKrolic** [comment]
`SA1515 Single-line comment should be preceded by a blank line` shouldn't probably be reported if the comment is above the first element of a multi-line collection expression:
```cs
string[] s =
[
    // This comment triggers SA1515
    "First element"
];
```

**Comment by jrmoreno1** [comment]
Not sure this is the right place to report it, but the line 

`return [.. common, .. specific];`

Triggers [SA1010](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1010.md), but the fix for that triggers [SA1000](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1000.md), which takes you back to the original code
